selinux: Update the policy for RHEL8

b-ranto opened the pull request:
We hit a couple more SELinux denials when running ceph on RHEL8. The
dac_read_search change is related to a kernel change where it checks
dac_read_search before dac_override, now.

Signed-off-by: Boris Ranto <branto@redhat.com>

This was reported downstream in

https://bugzilla.redhat.com/show_bug.cgi?id=1707762

<!--
Thank you for opening a pull request!  Here are some tips on creating
a well formatted contribution.

Please give your pull request a title like "[component]: [short description]"

This is the format for commit messages:

"""
[component]: [short description]

[A longer multiline description]

Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
Signed-off-by: [Your Name] <[your email]>
"""

The Signed-off-by line is important, and it is your certification that
your contributions satisfy the Developers Certificate or Origin.  For
more detail, see SubmittingPatches.rst.

The component is the short name of a major daemon or subsystem,
something like "mon", "osd", "mds", "rbd, "rgw", etc. For ceph-mgr modules,
give the component as "mgr/<module name>" rather than a path into pybind.

For more examples, simply use "git log" and look at some historical commits.

This was just a quick overview.  More information for contributors is available here:
https://raw.githubusercontent.com/ceph/ceph/master/SubmittingPatches.rst

-->

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

tchaikov commented on selinux/ceph.te:
```diff
@@ -119,6 +120,13 @@ nis_use_ypbind_uncond(ceph_t)
 storage_raw_rw_fixed_disk(ceph_t)
 files_manage_generic_locks(ceph_t)
 libs_exec_ldconfig(ceph_t)
+fs_list_hugetlbfs(ceph_t)
```
```
type=AVC msg=audit(1557729649.316:4222): avc:  denied  { getattr } for  pid=25994 comm="ms_dispatch" path="/dev/hugepages" dev="hugetlbfs" ino=21664 scontext=system_u:system_r:ceph_t:s0 tcontext=system_u:object_r:hugetlbfs_t:s0 tclass=dir permissive=1
type=SYSCALL msg=audit(1557729649.316:4222): arch=c000003e syscall=262 success=yes exit=0 a0=25 a1=55fd700e21bb a2=7f3774e652a0 a3=0 items=0 ppid=1 pid=25994 auid=4294967295 uid=167 gid=167 euid=167 suid=167 fsuid=167 egid=167 sgid=167 fsgid=167 tty=(none) ses=4294967295 comm="ms_dispatch" exe="/usr/bin/ceph-mon" subj=system_u:system_r:ceph_t:s0 key=(null)^]ARCH=x86_64 SYSCALL=newfstatat AUID="unset" UID="ceph" GID="ceph" EUID="ceph" SUID="ceph" FSUID="ceph" EGID="ceph" SGID="ceph" FSGID="ceph"
type=PROCTITLE msg=audit(1557729649.316:4222): proctitle=2F7573722F62696E2F636570682D6D6F6E002D66002D2D636C75737465720063657068002D2D6964006D61676E61303330002D2D736574757365720063657068002D2D73657467726F75700063657068
type=AVC msg=audit(1557729649.316:4223): avc:  denied  { read } for  pid=25994 comm="ms_dispatch" name="/" dev="hugetlbfs" ino=21664 scontext=system_u:system_r:ceph_t:s0 tcontext=system_u:object_r:hugetlbfs_t:s0 tclass=dir permissive=1
type=AVC msg=audit(1557729649.316:4223): avc:  denied  { open } for  pid=25994 comm="ms_dispatch" path="/dev/hugepages" dev="hugetlbfs" ino=21664 scontext=system_u:system_r:ceph_t:s0 tcontext=system_u:object_r:hugetlbfs_t:s0 tclass=dir permissive=1
```

are we using hugepage support in TCMalloc or using the DPDK backend?

b-ranto commented on selinux/ceph.te:
```diff
@@ -52,7 +53,7 @@ files_pid_file(ceph_var_run_t)
 allow ceph_t self:fifo_file rw_fifo_file_perms;
 allow ceph_t self:unix_stream_socket create_stream_socket_perms;
-allow ceph_t self:capability { setuid setgid dac_override };
+allow ceph_t self:capability { setuid setgid dac_override dac_read_search };
```
We have been doing this for quite some time. However, there was a change in kernel [1] and now, the kernel generates the `dac_read_search` even though it eventually grants the permission to read the file because of `dac_override` that we have already had in our policy for a while.

[1] https://danwalsh.livejournal.com/77140.html

tchaikov replied:
i've read the very same blog.

tchaikov commented on selinux/ceph.te:
```diff
@@ -119,6 +120,13 @@ nis_use_ypbind_uncond(ceph_t)
+fs_read_cgroup_files(ceph_t)
+fs_read_tmpfs_symlinks(ceph_t)
+fs_search_cgroup_dirs(ceph_t)
+init_nnp_daemon_domain(ceph_t)
```
```
ceph.te:128:ERROR 'syntax error' at token 'init_nnp_daemon_domain' on line 8624:
init_nnp_daemon_domain(ceph_t)
 	
/usr/bin/checkmodule:  error(s) encountered while parsing configuration
/usr/bin/checkmodule:  loading policy configuration from tmp/ceph.tmp
make[3]: *** [tmp/ceph.mod] Error 1
make[2]: *** [selinux/CMakeFiles/ceph.pp] Error 2
make[1]: *** [selinux/CMakeFiles/ceph.pp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

guess this depends on RHEL 8 policy settings. see https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos7,DIST=centos7,MACHINE_SIZE=huge/314//consoleFull

b-ranto commented:
@tchaikov Weirdly, that line works fine on my centos 7 machine. I could have a later selinux-policy package that introduced the function there, I suppose. I have repushed, removed the offending line and added the body of the offending function to the patch. A brief googling suggests that this should allow systemd to use a new mechanism to transition context of the processes that it runs.

tchaikov commented:
https://shaman.ceph.com/builds/ceph/wip-kefu-testing-2019-06-05-1253/5722c6457fd0ba2d97fa4a7b0422622961ab38a8/

tchaikov commented:
```
ceph.te:145:ERROR 'unknown class process2' at token ';' on line 8723:
allow ceph_t init_var_run_t:file getattr;
allow init_t ceph_t:process2 { nnp_transition nosuid_transition };
/usr/bin/checkmodule:  error(s) encountered while parsing configuration
/usr/bin/checkmodule:  loading policy configuration from tmp/ceph.tmp
make[3]: *** [tmp/ceph.mod] Error 1
make[2]: *** [selinux/CMakeFiles/ceph.pp] Error 2
make[1]: *** [selinux/CMakeFiles/ceph.pp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos7,DIST=centos7,MACHINE_SIZE=huge/402//consoleFull

b-ranto commented:
I have added

`class process2 { nnp_transition nosuid_transition };`

line to the requires section. that should (hopefully) help with the build failure.

tchaikov commented:
https://shaman.ceph.com/builds/ceph/wip-kefu-testing-2019-06-12-0017/3ecb65aee2c3bc1a52cd9bd0699b6145f742f15d/
